coreapps-web-2.2/src/main/java/org/openmrs/module/coreapps/diagnosisCompatibility/DiagnosisServiceCompatibility2_2.java

```diff
@@ -0,0 +1,23 @@
+package org.openmrs.module.coreapps.diagnosisCompatibility;
+
+import java.util.List;
+
+import org.openmrs.Diagnosis;
+import org.openmrs.Encounter;
+import org.openmrs.api.DiagnosisService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+
+public class DiagnosisServiceCompatibility2_2 implements DiagnosisServiceCompatibility  {
+	
+	@Autowired
+	@Qualifier("diagnosisService")
+	DiagnosisService diagnosisService;
```

Comment:
ideally , By standard we acces OpenMRS services through the Context , not sure why you decided to Autowire it

Reply:
@mozzy11  so i did this since using Context also returns the same error

Reply:
@HerbertYiga make this `private`.